boot-passthrough.sh

```diff
@@ -1,22 +1,34 @@
 #!/usr/bin/env bash
 
+ALLOCATED_RAM="3072" # MiB
+CPU_SOCKETS="1"
+CPU_CORES="2"
+CPU_THREADS="4"
+
+REPO_PATH="./"
+OVMF_DIR="."
+
 # shellcheck disable=SC2054
 args=(
-  -enable-kvm -m 3072 -cpu Penryn,kvm=on,vendor=GenuineIntel,+invtsc,vmware-cpuid-freq=on
+  -enable-kvm -m "$ALLOCATED_RAM" -cpu host,kvm=on,vendor=GenuineIntel,+invtsc,vmware-cpuid-freq=on
   -machine pc-q35-2.9
-  -smp 4,cores=2
+  -smp "$CPU_THREADS",cores="$CPU_CORES",sockets="$CPU_SOCKETS"
```

Comment:
This is misleading, the `4` was a number of cpu not number of threads, it can be set by plain number like `4` or `cpu=4`. You may also use `cpus="$(nproc)"` instead of hardcoding specific value.  Threads should be set with `threads=x`, see qemu docs.

Side note: qemu devs don't recommend splitting topology between cores and threads but raise the number of cores instead which should give better stability. I think setting cores to same number as cpus and leaving threads and sockets with `1` would be best default.

Reply:
@kholia any opinion on this? @Maryse47 has a point because I did experience stability issues when using `smp=12, core=6,socket=1` but no issues with `smp=12` alone but I simply did the changes in line with what other macOS KVM guides that seem to explicitly mention in terms of topology mentions so I didn't change much.

I would advise against `cpus="$(nproc)"`, some users run multiple VMs (and use a program like Syngery for cross linking) or run additional programs within the hypervisor-hosting OS (e.g. I ran a Samba share that gave me access to a common ext4 partition between all VMs) so I think that default allocation of all available threads shouldn't be hardcoded.

Reply:
I don't know about this topic. Any of the options which work reliably are fine with me.

Reply:
Why it was merged then? The `CPU_THREADS` variable name is obviously wrong.